### PR TITLE
fix font render for modified fonts

### PR DESCRIPTION
**Pull request by xlmnxp**
Signed-off-by: Salem Yaslem <s@sy.sa>

I discuses about this issue on Snapcraft forum: 
https://forum.snapcraft.io/t/snap-package-cannot-read-fonts-conf/

### CODE REVIEW

**Comment by oSoMoN** [comment]
That looks fine to me. Can you edit the comment to be more specific about what this allows? And add a link to the forum thread, for future reference.

@jhenstridge would you mind validating this change, since you have a deeper understanding of the snap/fontconfig subtle interactions?

**Comment by xlmnxp** [comment]
> That looks fine to me. Can you edit the comment to be more specific about what this allows? And add a link to the forum thread, for future reference.
> 
> @jhenstridge would you mind validating this change, since you have a deeper understanding of the snap/fontconfig subtle interactions?

I will do that

**Comment by xlmnxp** [comment]
@oSoMoN you can check it now

**Comment by xlmnxp** [comment]
@oSoMoN @jhenstridge 
Can anyone review the PR?

File: common/desktop-exports [hunk 1]
```diff
@@ -289,6 +289,10 @@ function make_user_fontconfig {
   if [ -d $REALHOME/.cache/fontconfig ]; then
     echo "  <cachedir>$REALHOME/.cache/fontconfig</cachedir>"
   fi
+  
+  # fix font render for modified fonts, that discussed on Snapcraft Forum:
+  # https://forum.snapcraft.io/t/snap-package-cannot-read-fonts-conf/16657
+  echo '  <include ignore_missing="yes">/etc/fonts/conf.d</include>'
```
Comment (ajvengo) [inline common/desktop-exports:295]: This line should go before `echo '  <include ignore_missing="yes">conf.d</include>'` to allow user to override `global` config by `user` specific one

Comment (xlmnxp) [reply common/desktop-exports]: Thank you, I resolve it :smile:

**Comment by xlmnxp** [comment]
Will it be merge or there something want to change in the code?

**Comment by vibraniumdev** [comment]
Is this PR being merged? I just stumbled into this missing fonts error in my Todoist (snap)

File: common/desktop-exports [hunk 2]
```diff
@@ -281,15 +281,19 @@ function make_user_fontconfig {
...
+  
   # We need to include this default cachedir first so that caching
   # works: without it, fontconfig will try to write to the real user home
   # cachedir and be blocked by AppArmor.
   echo '  <cachedir prefix="xdg">fontconfig</cachedir>'
   if [ -d $REALHOME/.cache/fontconfig ]; then
     echo "  <cachedir>$REALHOME/.cache/fontconfig</cachedir>"
-  fi
-  echo "</fontconfig>"
+  fi
```
Comment (diddledani) [inline common/desktop-exports:295]:
```suggestion
  fi
```

File: common/desktop-exports [hunk 3]
```diff
@@ -281,15 +281,19 @@ function make_user_fontconfig {
...
   # We need to include this default cachedir first so that caching
   # works: without it, fontconfig will try to write to the real user home
   # cachedir and be blocked by AppArmor.
   echo '  <cachedir prefix="xdg">fontconfig</cachedir>'
   if [ -d $REALHOME/.cache/fontconfig ]; then
     echo "  <cachedir>$REALHOME/.cache/fontconfig</cachedir>"
-  fi
-  echo "</fontconfig>"
+  fi 
+ echo "</fontconfig>"
```
Comment (diddledani) [inline common/desktop-exports:296]:
```suggestion
  echo "</fontconfig>"
```

File: common/desktop-exports [hunk 4]
```diff
@@ -281,15 +281,19 @@ function make_user_fontconfig {
       echo "  <dir>${data_dirs_array[$i]}/fonts</dir>"
     fi
   done
+  # fix font render for modified fonts, that discussed on Snapcraft Forum:
+  # https://forum.snapcraft.io/t/snap-package-cannot-read-fonts-conf/16657
+  echo '  <include ignore_missing="yes">/etc/fonts/conf.d</include>'
```
Comment (diddledani) [inline common/desktop-exports:286]: Does this rule expect to use the `fonts/conf.d` from the base snaps, or are you expecting it to read from the host?

If you are expecting it to read from the host then it won't work as expected.

Comment (diddledani) [reply common/desktop-exports:286]: Ignore this, I just double checked - the directory is correctly accessible inside snaps. 🤦🏻‍♀️

Comment (xlmnxp) [reply common/desktop-exports:286]: so it not need any edit?

Comment (diddledani) [reply common/desktop-exports:286]: yup, the PR is good to go now :-)

**Review by diddledani — APPROVED** [review]

**Pull request merged**